openfe/setup/box_representation.py

```diff
@@ -0,0 +1,104 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+import numpy as np
+
+
+def is_valid_box(box: np.ndarray) -> bool:
```

Comment:
Is it our job to check the validity of the box vectors? Different MD engines will fail for different kinds of boxes. You can create boxes that pass the tests here and aren't valid for OpenMM. You can create boxes that are valid for Gromacs and aren't valid for OpenMM.

Reply:
That is a good question, I think we shouldn't evaluate if an engine supports something until we hand it off to the engine, that way a user can defer engine choice until they absolutely need to and then we can have each engine wrapper handle validation instead of having validation checks in each of our container classes

Reply:
+1 to both the above